hazelcast/src/main/java/com/hazelcast/query/impl/IndexImpl.java

```diff
@@ -40,16 +41,18 @@
     public static final NullObject NULL = new NullObject();
 
     // indexKey -- indexValue
-    private final ConcurrentMap<Data, Comparable> recordValues = new ConcurrentHashMap<Data, Comparable>(1000);
     private final IndexStore indexStore;
     private final String attribute;
     private final boolean ordered;
 
     private volatile TypeConverter converter;
 
-    public IndexImpl(String attribute, boolean ordered) {
+    private SerializationService ss;
+
+    public IndexImpl(String attribute, boolean ordered, SerializationService ss) {
         this.attribute = attribute;
         this.ordered = ordered;
+        this.ss = ss;
         indexStore = (ordered) ? new SortedIndexStore() : new UnsortedIndexStore();
```

Comment:
why are there (...) around ordered? It is not your code, but this doesn't make any sense.. it just adds reading complexity.